code/frameworks/nextjs/src/headers/headers.ts

```diff
@@ -0,0 +1,104 @@
+import { fn } from '@storybook/test';
+import type { IncomingHttpHeaders } from 'http';
+
+// Mostly copied from https://github.com/vercel/next.js/blob/763b9a660433ec5278a10e59d7ae89d4010ba212/packages/next/src/server/web/spec-extension/adapters/headers.ts#L20
+export class HeadersStore extends Headers {
+  private readonly headers: IncomingHttpHeaders = {};
+
+  /** @internal */
+  mockRestore = () => {
+    this.forEach((key) => this.delete(key));
+  };
+
+  /**
+   * Merges a header value into a string. This stores multiple values as an
+   * array, so we need to merge them into a string.
+   *
+   * @param value a header value
+   * @returns a merged header value (a string)
+   */
+  private merge(value: string | string[]): string {
+    if (Array.isArray(value)) return value.join(', ');
+
+    return value;
+  }
+
+  public append = fn((name: string, value: string): void => {
+    const existing = this.headers[name];
+    if (typeof existing === 'string') {
+      this.headers[name] = [existing, value];
+    } else if (Array.isArray(existing)) {
+      existing.push(value);
+    } else {
+      this.headers[name] = value;
+    }
+  }).mockName('headers().append');
+
+  public delete = fn((name: string) => {
+    delete this.headers[name];
+  }).mockName('headers().delete');
+
+  public get = fn((name: string): string | null => {
```

Comment:
What if I do `headers().get('Cookie')`? Will it get the mocked cookie value from the cookie store?

Reply:
yes, but `headers().get('cookie')` instead

Reply:
headers are case insensitive I belief

Reply:
<img width="440" alt="image" src="https://github.com/storybookjs/storybook/assets/1035299/ca16d80a-eb9a-45e4-95b6-abe1278e87c1">

Reply:
actually thinking about it, why do extend `Headers`? We override everything no?

Reply:
Because the Next Headers do the same, there are other fundamental methods that would need to be implemented like getSetCookie and a couple others

Reply:
right, makes sense